.github/workflows/main.yml

```diff
@@ -174,12 +174,15 @@ jobs:
   memcheck:
     runs-on: ubuntu-latest
     steps:
+    - uses: actions/checkout@v3
     - name: Cache valgrind
       uses: actions/cache@v3
+      id: cache-valgrind
       with:
         path: valgrind-3.20.0
         key: ${{ runner.os }}-valgrind-3.20.0
-    - name: Install valgrind from source
+    - name: Download valgrind
+      if: steps.cache-valgrind.outputs.cache-hit != 'true'
```

Comment:
This will mean `libc6-dbg` might not be installed (on cache hit).
Is it only needed for `./configure` and `make`, or also at runtime?
If the latter, I think best to just move the apt-get stuff to another step.

Reply:
Yeah I ended up having to. I thought it was possible the built artifact would contain the necessary functions, but apparently not.